annotations/static/annotations/js/annotators/text.js

```diff
@@ -327,24 +330,123 @@ DateAppellationCreator = {
         }
     }
 }
+ConceptPickerItem = {
+    props: ['concept'],
+    components: {
+    },
+    template: `<div class="list-group-item concept-item clearfix" id="concept-{{ concept.interpretation.uri }}">
+                <div>
+                    <a v-on:click="select" style="cursor: pointer;">{{ concept.interpretation_label }} ({{ concept.interpretation.authority }})</a>
+                </div>
+                <div class="text text-muted">{{ concept.interpretation.description }}</div>
+            </div>`,
+    methods: {
+        select: function() {
+            this.$emit('selectconcept', this.concept);
+            },
 
+        }
+    }
+
+ConceptPicker = {
+    props: ['appellations'],
+    components: {
+        'concept-picker-item': ConceptPickerItem
+    },
+    data: function() {
+        return {
+            conceptsFinal: [],
+            appell: [],
+            appellationCount: [],
+            count: 0
+        }
+    },
+    template: `<div  class="concept-picker" style="max-height: 50vh; overflow-y: scroll;">
+                <concept-picker-item
+                    v-on:selectconcept="selectConcept"
+                    v-for="concept in conceptsFinal"
+                    v-bind:concept=concept>
+                </concept-picker-item>
+               </div>`,
+    methods: {
+        selectConcept: function(concept) {
+            // Clear the concept search results.
+            this.concepts = [];
+            this.$emit('selectconcept', concept);
+        },
+        addConcepts: function (appellationMapEntires) {
+            while(this.count <= 3) {
+                var appellation = appellationMapEntires.next().value;
+                if (appellation == null) {
+                    break
+                } 
+                if (!this.conceptsFinal.includes(appellation[1][0])){
+                    this.conceptsFinal.push(appellation[1][0]);
+                    this.count++;
+                }
+            }
+            // reset count
+            this.count = 0;
```

Comment:
is there a reason that this is not a local variable?

Reply:
Nope, I guess I just enjoy putting 'this' in front of everything.